clippy_lints/src/eta_reduction.rs

```diff
@@ -32,7 +32,26 @@ declare_clippy_lint! {
     "redundant closures, i.e., `|a| foo(a)` (which can be written as just `foo`)"
 }
 
-declare_lint_pass!(EtaReduction => [REDUNDANT_CLOSURE]);
+declare_clippy_lint! {
+    /// **What it does:** Checks for closures which only invoke a method on the closure
+    /// argument and can be replaced by referencing the method directly.
+    ///
+    /// **Why is this bad?** It's unnecessary to create the closure.
+    ///
```

Comment:
Could you add `Known issues` and explain there why it was separated from `redundant_closure`?